### PR TITLE
ENG-2034 Implement deleting components

### PR DESCRIPTION
**Pull request by SmartMonkey-git**
[Task](https://datarevenue.atlassian.net/browse/ENG-2034)

### CODE REVIEW

File: tycho-indexer/src/storage/postgres/protocol.rs [hunk 1]
```diff
@@ -642,4 +673,51 @@ mod test {
...
+            .iter()
+            .collect::<Vec<_>>();
+
+        let res = gw
+            .delete_protocol_components(&to_delete, &mut conn)
+            .await;
+
+        assert!(res.is_ok());
+        for (index, component) in test_components.iter().enumerate() {
+            let updated_ts = schema::protocol_component::table
```
Comment (tamaralipows) [inline tycho-indexer/src/storage/postgres/protocol.rs]: Is this the `updated_ts`? It seems to be the `deleted_at`

Comment (SmartMonkey-git) [reply tycho-indexer/src/storage/postgres/protocol.rs]: It is the deleted_at timestamp that got updated.
The column that indicates an update of the table is called modified_ts

File: tycho-indexer/src/storage/postgres/protocol.rs [hunk 2]
```diff
@@ -642,4 +673,51 @@ mod test {
...
+                .filter(schema::protocol_component::external_id.eq(component.id.0.to_string()))
+                .select(schema::protocol_component::deleted_at)
+                .first::<Option<NaiveDateTime>>(&mut conn)
+                .await
+                .unwrap();
+
+            assert!(updated_ts.is_some());
```
Comment (tamaralipows) [inline tycho-indexer/src/storage/postgres/protocol.rs]: Should we also check the length here? I think we expect 2, no?

File: tycho-indexer/src/storage/postgres/protocol.rs [hunk 3]
```diff
@@ -149,6 +150,25 @@ where
...
+            .iter()
+            .map(|c| c.id.0.to_string())
+            .collect();
+        let current_time: NaiveDateTime = Utc::now().naive_utc();
+
+        diesel::update(protocol_component.filter(external_id.eq_any(ids_to_delete)))
+            .set(deleted_at.eq(current_time))
+            .execute(conn)
+            .await?;
+        Ok(())
```
Comment (dianacarvalho1) [inline tycho-indexer/src/storage/postgres/protocol.rs]: this is the delete behaviour we want?? 🤔 why not real delete them?

Comment (kayibal) [reply tycho-indexer/src/storage/postgres/protocol.rs]: Backtesting

File: tycho-indexer/src/storage/postgres/protocol.rs [hunk 4]
```diff
@@ -149,6 +150,25 @@ where
...
+        use super::schema::protocol_component::dsl::*;
+
+        let ids_to_delete: Vec<String> = to_delete
+            .iter()
+            .map(|c| c.id.0.to_string())
+            .collect();
+        let current_time: NaiveDateTime = Utc::now().naive_utc();
+
+        diesel::update(protocol_component.filter(external_id.eq_any(ids_to_delete)))
+            .set(deleted_at.eq(current_time))
```
Comment (kayibal) [inline tycho-indexer/src/storage/postgres/protocol.rs]: This can't be current_time it must be related to a block timestamp

**Pull request merged**